pkg/apis/extensions/v1beta1/conversion.go

```diff
@@ -27,6 +27,13 @@ import (
 	"k8s.io/kubernetes/pkg/conversion"
 	"k8s.io/kubernetes/pkg/runtime"
 	"k8s.io/kubernetes/pkg/util/intstr"
+	"k8s.io/kubernetes/pkg/util/labels"
+	"k8s.io/kubernetes/pkg/util/validation/field"
+)
+
+const (
+	// annotation key used to identify non-convertible json paths.
+	NonConvertibleAnnotation = "kubernetes.io/non-convertible"
```

Comment:
I think I'd like to see this as a prefix.  We can do something like `prefix + "/" + field.Path` and have it equal the old value for blind round tripping.